### ISSUE TITLE
[FEATURE] wash CTL commands shouldn't require a host ID if there's only one running

### ISSUE THREAD

**Issue by brooksmtownsend** [wasmCloud/wasmCloud#3259]
### Affected project(s)

- [ ] documentation
- [ ] examples
- [ ] wasmCloud host
- [X] wasmCloud CLI (wash)
- [ ] wasmCloud dashboard UI (washboard)
- [ ] capability providers
- [ ] provider bindgen
- [ ] control interface client
- [ ] other / not sure

### Is your feature request related to a problem? Please describe.

The current `scale` command is as follows:
```
➜ wash scale component
error: the following required arguments were not provided:
  <host-id>
  <component-ref>
  <component-id>

Usage: wash scale component <host-id> <component-ref> <component-id>
```

Even if I'm only running one host, I always have to specify a host ID.

### Describe the solution you'd like

If I'm only running one host, e.g. for local development, I should be able to lean on the existing feature to avoid specifying a host ID for all control interface commands that "require" one if there's only one host present. This would include `wash stop host`, etc.

### Additional context

See the `wash get inventory` command for one way we could approach this today.

Related to #3258, since we may be able to re-use host lookup code

**Comment by Waterdrips** [comment]
Can I take a look at this - just a few questions though - looking specifically at scale:

First off, how do you see us making host-id optional considering it's the first arg? We can't simply make it optonal as those optional args need to be after required args.

We can't really change the signature as host-id is required now so moving it to the end would be a breaking change? 

We could do some funky arg-parsing (make the last one there optional, then check if we have 2 args, if we do just assume they are the non-host one and continue.)

I got this "working" with putting host-id last, if its not set passing in "" to the command 

```sh
heyal@heyal-desktop:~/code/wasmCloud/target/debug/new$ ../wash scale component /home/heyal/code/wasmCloud/target/debug/new/build/http_hello_world_s.wasm new

Component [new] (ref: [file:///home/heyal/code/wasmCloud/target/debug/new/build/http_hello_world_s.wasm]) scaled on host [NDRHCICMAAQBBK2RLE4NC76L2ATBB3RRQSVDLVRH44IME7TPTUMS5BOS] to unbounded concurrency

```

**Comment by markkovari** [comment]
Hey @Waterdrips, first of all you are absolutely correct with the breaking change, but the targeted "artifact" `wash-cli` is pre 1.x.x, so breaking changes are acceptable afaik. Did/thought the same by moving the host-id to the end, but had some issues with the testing.
I was not really able to create multiple TestWashInstances on the same lattice in the integration tests.

The other command which is interesting is the label with the key--value pairs, you need probably custom parser there, I just kept the hostid there 😅 

Did not ask to assign the issue to me since I have something else on me and I am on and off lately, did not want to squat on it.

But if you would see my [partial progress](https://github.com/wasmCloud/wasmCloud/compare/main...markkovari:wasmCloud:feat/host-id-can-be-omitted-if-only-one-host-is-present?expand=1).

Definitely did want to yoink the issue before others, just played with the idea/solution.

**Comment by Waterdrips** [comment]
ha yeh `label` will be interesting. Easy enough when its not a `--delete`. 

On delete you only need to pass the keys so how do you decide which one to pass as the host (if any?) - try the last one? if it's not a valid host then call the `find_host_id` function with nothing and see if there's only one host? :thinking:

**Comment by brooksmtownsend** [comment]
Hey @Waterdrips ! I think it seems like some of these changes are going to be easier than others, `host-id` being the first argument is likely to be the worst breaking change to make of these, which I'd consider a phased replacement:
1. Consider the current `wash` version to be `0.36.0`
2. In wash `0.37.0`, we mark the `<host-id>` argument as deprecated and introduce a `--host-id` flag which should be preferred
3. In wash `0.38.0`, we transition to fully use just the `--host-id` flag and remove the argument

This would give users a full minor version to transition scripts, and we can do step "2" at any time to enhance backwards compatibility. Thoughts? No matter where the `host-id` argument is present, I think it may be a good idea to standardize on the flag to help the transition here.

Last note, as you mentioned with `label`, if there are any commands that really don't fit this issue, feel free to call them out and skip them

**Comment by brooksmtownsend** [comment]
> so for now I can add a host-id flag to the commands which have it as an arg only and write tests for this

Yes this sounds great! Perfect thing to get released first as a minor version, along with a warning log to note that the positional argument for host ID is deprecated. I think the flag taking precedent would make sense as it's new and will be the only way to supply it in the long term
